core/src/main/java/org/apache/accumulo/core/singletons/SingletonReservation.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.core.singletons;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @see SingletonManager#getClientReservation()
+ */
+public class SingletonReservation implements AutoCloseable {
+
+  // volatile so finalize does not need to synchronize to reliably read
+  protected volatile boolean closed = false;
+
+  private static Logger log = LoggerFactory.getLogger(SingletonReservation.class);
+
+  @Override
+  public synchronized void close() {
+    if (closed) {
+      return;
+    }
+    closed = true;
+    SingletonManager.releaseRerservation();
+  }
+
+  @Override
+  protected void finalize() throws Throwable {
+    try {
+      if (!closed) {
+        log.warn("An Accumulo Client was garbage collected without being closed.");
```

Comment:
Could this become a burden on performance?  If so, then do we really need to warn about this?

Reply:
I don't think this would cause problems since its once per client.  As for logging, I was uncertain about adding it.  However I convinced myself it will be useful for peoples initial transition.  We can always remove this code later w/o causing any change in runtime behavior of the API.